### PR TITLE
Download and Export CSV

### LINKED ISSUE #25: Export Data
Make an exportable .csv file from the database.

- [x] Export button on location page
- [x] Convert data (date range specified??) to csv
- [x] Open share tray

### PR DESCRIPTION
**Pull request by HarperThurlow**
This PR allows the user to download .csv file of the SingleWeatherReport and make it possible to share using Android Share Sheet.

Currently the file location is hard-coded to appear in our folder for the application. This can be changed upon a later review or if we want it saved somewhere else (However must be coded and is not dynamic).

If reviewers can please checkout this branch to test if everything is running smoothly that would be great :)

### CODE REVIEW

File: app/src/main/java/com/cp3407/wildernessweather/ExportCSV.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import org.supercsv.prefs.CsvPreference;
+
+import java.io.FileWriter;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+public class ExportCSV {
+    /*
```
Comment (JackMcGill) [inline app/src/main/java/com/cp3407/wildernessweather/ExportCSV.java:12]: Maybe move this link to the ExportCSV documentation page, with a brief description of the `super-csv` dependancy.
**Edit-** I'm actually referencing the link on line `17`, looks like I have highlighted the wrong lines in this comment.

**Review by JackMcGill — CHANGES_REQUESTED** [review]
I think this PR is _almost_ ready to go. Before it's merged into `develop` however I think that we should do 2 things:

1. The button icon used to create and share the CSV file is currently a "download" button. I think this should be changed to a generic "Share" icon, as I feel this better suits the action that takes place when this button is pressed.
2. There have been updates to the properties inside `WeatherReportModel`, specifically there is now a `woeid` property. Therefore the code that creates the CSV should probably be updated to reflect this. This can be done by either rebasing this branch onto `develop`, or perhaps an easier solution, is to just replace the contents of `WeatherReportModel` with the version on the `develop` branch.

Both these changes are minor and I'm happy to help implement both if you agree with them.
Otherwise everything else looks good to me, and is otherwise ready to merge 😄

File: app/src/main/java/com/cp3407/wildernessweather/ExportCSV.java (hunk 1 above)
Comment (HarperThurlow) [reply app/src/main/java/com/cp3407/wildernessweather/ExportCSV.java:12]: Ya documentation does need a bit of work for sure. I'll throw a small talk about the dependency now :)

**Review by calebwebster — APPROVED** [review]
Looks good.

File: app/src/main/java/com/cp3407/wildernessweather/ExportCSV.java (hunk 1 above)
Comment (JackMcGill) [reply app/src/main/java/com/cp3407/wildernessweather/ExportCSV.java:12]: 👍

**Review by JackMcGill — APPROVED** [review]
Good to go now.

**Pull request merged**